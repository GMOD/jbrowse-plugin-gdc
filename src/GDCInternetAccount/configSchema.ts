import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { Instance } from 'mobx-state-tree'
import PluginManager from '@jbrowse/core/PluginManager'
import { createInternetAccountConfig } from '@jbrowse/core/pluggableElementTypes/models'

function GDCConfigFactory(pluginManager: PluginManager) {
  return ConfigurationSchema(
    'GDCInternetAccount',
    {
      validDomains: {
        description:
          'array of valid domains the url can contain to use this account. Empty = all domains',
        type: 'stringArray',
        defaultValue: [],
      },
      authHeader: {
        description: 'custom auth header for authorization',
        type: 'string',
        defaultValue: 'Authorization',
      },
      customEndpoint: {
        description: 'custom endpoint for the external token resource',
        type: 'string',
        defaultValue: '',
      },
    },
    {
      baseConfiguration: createInternetAccountConfig(pluginManager),
      explicitlyTyped: true,
    },
  )
}

export type GDCInternetAccountConfigModel = ReturnType<typeof GDCConfigFactory>
export type OAuthInternetAccountConfig = Instance<GDCInternetAccountConfigModel>
export default GDCConfigFactory
