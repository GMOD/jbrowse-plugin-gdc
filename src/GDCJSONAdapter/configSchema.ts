import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { types } from 'mobx-state-tree'

export default ConfigurationSchema(
  'GDCJSONAdapter',
  {
    featureType: {
      type: 'stringEnum',
      model: types.enumeration('Feature Type', ['mutation', 'gene']),
      defaultValue: 'mutation',
      description: 'The type of the track to add',
    },
    data: {
      type: 'string',
      defaultValue: '',
      description: 'JSON string of the GDC data from the file'
    }
  },
  { explicitlyTyped: true },
)
