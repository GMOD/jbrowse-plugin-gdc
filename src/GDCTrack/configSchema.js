import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export default pluginManager => {
    const { BaseTrackConfig: LinearGenomeTrackConfig }  = pluginManager.getPlugin('LinearGenomeViewPlugin').exports
    return ConfigurationSchema(
    'GDCTrack',
    {
      adapter: pluginManager.pluggableConfigSchemaType('adapter'),
      renderer: pluginManager.pluggableConfigSchemaType('renderer'),
    },
    { baseConfiguration: LinearGenomeTrackConfig, explicitlyTyped: true },
  )
}
