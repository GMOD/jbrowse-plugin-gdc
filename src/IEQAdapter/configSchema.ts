import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default ConfigurationSchema(
  'IeqAdapter',
  {
    ieqLocation: {
      type: 'fileLocation',
      defaultValue: { uri: '/path/to/myfile.tsv', locationType: 'UriLocation' },
    },
  },
  { explicitlyTyped: true },
)
