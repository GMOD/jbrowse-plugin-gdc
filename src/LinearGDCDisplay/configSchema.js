import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default pluginManager => {
  const { baseLinearDisplayConfigSchema } = pluginManager.getPlugin(
    'LinearGenomeViewPlugin',
  ).exports
  return ConfigurationSchema(
    'LinearGDCDisplay',
    { renderer: pluginManager.pluggableConfigSchemaType('renderer') },
    { baseConfiguration: baseLinearDisplayConfigSchema, explicitlyTyped: true },
  )
}
