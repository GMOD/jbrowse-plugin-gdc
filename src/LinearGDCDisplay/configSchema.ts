import PluginManager from '@jbrowse/core/PluginManager'
import { ConfigurationSchema } from '@jbrowse/core/configuration'
import LinearGenomeViewPlugin from '@jbrowse/plugin-linear-genome-view'

export default (pluginManager: PluginManager) => {
  const { baseLinearDisplayConfigSchema } = (
    pluginManager.getPlugin('LinearGenomeViewPlugin') as LinearGenomeViewPlugin
  ).exports
  return ConfigurationSchema(
    'LinearGDCDisplay',
    { renderer: pluginManager.pluggableConfigSchemaType('renderer') },
    { baseConfiguration: baseLinearDisplayConfigSchema, explicitlyTyped: true },
  )
}
