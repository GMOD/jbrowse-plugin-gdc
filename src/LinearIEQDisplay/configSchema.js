import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default pluginManager => {
  const { baseLinearDisplayConfigSchema } = pluginManager.getPlugin(
    'LinearGenomeViewPlugin',
  ).exports
  return ConfigurationSchema(
    'LinearIEQDisplay',
    { renderer: pluginManager.pluggableConfigSchemaType('renderer') },
    { baseConfiguration: baseLinearDisplayConfigSchema, explicitlyTyped: true },
  )
}
