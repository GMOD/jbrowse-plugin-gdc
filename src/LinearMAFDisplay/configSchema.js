import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default pluginManager => {
  const { baseLinearDisplayConfigSchema } = pluginManager.getPlugin(
    'LinearGenomeViewPlugin',
  ).exports
  return ConfigurationSchema(
    'LinearMAFDisplay',
    { renderer: pluginManager.pluggableConfigSchemaType('renderer') },
    { baseConfiguration: baseLinearDisplayConfigSchema, explicitlyTyped: true },
  )
}
