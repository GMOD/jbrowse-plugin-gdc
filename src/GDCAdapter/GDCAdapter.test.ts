import { toArray } from 'rxjs/operators'
import GDCAdapter from './GDCAdapter'
import configSchema from './configSchema'
import fetchMock from 'jest-fetch-mock'

fetchMock.enableMocks()

test('adapter can fetch features from the gdc', async () => {
  const adapter = new GDCAdapter(
    configSchema.create({
      filters: JSON.stringify({
        op: 'and',
        content: [
          {
            op: 'in',
            content: {
              field: 'cases.project.project_id',
              value: ['TCGA-UCEC'],
            },
          },
          {
            op: 'in',
            content: {
              field: 'genes.is_cancer_gene_census',
              value: ['true'],
            },
          },
          {
            op: 'in',
            content: {
              field: 'ssms.consequence.transcript.annotation.vep_impact',
              value: ['high'],
            },
          },
        ],
      }),
      //'{"op":"and","content":[{"op":"in","content":{"field":"cases.project.project_id","value":["TCGA-UCEC"]}},{"op":"in","content":{"field":"genes.is_cancer_gene_census","value":["true"]}},{"op":"in","content":{"field":"ssms.consequence.transcript.annotation.vep_impact","value":["high"]}}]}',
    }),
  )

  fetchMock.mockResponseOnce(
    JSON.stringify({
      data: {
        viewer: {
          explore: {
            features: {
              hits: {
                edges: [
                  {
                    node: {
                      chromosome: 'chr3',
                      consequence: {},
                      cosmicId: null,
                      endPosition: 124908333,
                      genomicDnaChange: 'chr3:g.124908333C>T',
                      mutationSubtype: 'Single base substitution',
                      mutationType: 'Simple Somatic Mutation',
                      ncbiBuild: 'GRCh38',
                      numOfCasesInCohort: 12922,
                      occurrenceInCohort: '5 / 12922',
                      percentage: 113.97940008672037,
                      referenceAllele: 'C',
                      score: 5,
                      ssmId: 'fe913ae3-e63e-585c-84cf-f47fcf51b639',
                      startPosition: 124908333,
                    },
                  },
                  {
                    node: {
                      chromosome: 'chr3',
                      consequence: {},
                      cosmicId: null,
                      endPosition: 124908160,
                      genomicDnaChange: 'chr3:g.124908160C>T',
                      mutationSubtype: 'Single base substitution',
                      mutationType: 'Simple Somatic Mutation',
                      ncbiBuild: 'GRCh38',
                      numOfCasesInCohort: 12922,
                      occurrenceInCohort: '2 / 12922',
                      percentage: 106.02059991327963,
                      referenceAllele: 'C',
                      score: 2,
                      ssmId: '68e512b9-511f-58e3-8f41-137c6c0291af',
                      startPosition: 124908160,
                    },
                  },
                  {
                    node: {
                      chromosome: 'chr3',
                      consequence: {},
                      cosmicId: null,
                      endPosition: 124908195,
                      genomicDnaChange: 'chr3:g.124908195C>A',
                      mutationSubtype: 'Single base substitution',
                      mutationType: 'Simple Somatic Mutation',
                      ncbiBuild: 'GRCh38',
                      numOfCasesInCohort: 12922,
                      occurrenceInCohort: '1 / 12922',
                      percentage: 100,
                      referenceAllele: 'C',
                      score: 1,
                      ssmId: '0198e63f-185f-526f-96e5-f54010ffd2c0',
                      startPosition: 124908195,
                    },
                  },
                ],
                total: 30,
              },
            },
            filteredCases: {
              hits: {
                total: 12922,
              },
            },
          },
        },
      },
    }),
  )

  const features = adapter.getFeatures({
    assemblyName: 'volvox',
    refName: 'chr3',
    start: 124900600,
    end: 124925000,
  })

  const names = await adapter.getRefNames()
  expect(names).toMatchSnapshot()

  const featuresArray = await features.pipe(toArray()).toPromise()
  expect(featuresArray.slice(0, 3)).toMatchSnapshot()
})
