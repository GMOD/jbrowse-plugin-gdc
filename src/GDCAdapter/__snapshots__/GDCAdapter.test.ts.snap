// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adapter can fetch features from the gdc 1`] = `
Array [
  "chr1",
  "chr10",
  "chr11",
  "chr12",
  "chr13",
  "chr14",
  "chr15",
  "chr16",
  "chr17",
  "chr18",
  "chr19",
  "chr2",
  "chr20",
  "chr21",
  "chr22",
  "chr3",
  "chr4",
  "chr5",
  "chr6",
  "chr7",
  "chr8",
  "chr9",
  "chrX",
  "chrY",
]
`;

exports[`adapter can fetch features from the gdc 2`] = `
Array [
  Object {
    "chromosome": "chr3",
    "consequence": Object {},
    "cosmicId": null,
    "end": 124908333,
    "endPosition": 124908333,
    "genomicDnaChange": "chr3:g.124908333C>T",
    "mutationSubtype": "Single base substitution",
    "mutationType": "Simple Somatic Mutation",
    "ncbiBuild": "GRCh38",
    "numOfCasesInCohort": 12922,
    "occurrenceInCohort": "5 / 12922",
    "percentage": 113.97940008672037,
    "refName": "chr3",
    "referenceAllele": "C",
    "score": 5,
    "ssmId": "fe913ae3-e63e-585c-84cf-f47fcf51b639",
    "start": 124908332,
    "startPosition": 124908333,
    "type": "Simple Somatic Mutation",
    "uniqueId": "fe913ae3-e63e-585c-84cf-f47fcf51b639",
  },
  Object {
    "chromosome": "chr3",
    "consequence": Object {},
    "cosmicId": null,
    "end": 124908160,
    "endPosition": 124908160,
    "genomicDnaChange": "chr3:g.124908160C>T",
    "mutationSubtype": "Single base substitution",
    "mutationType": "Simple Somatic Mutation",
    "ncbiBuild": "GRCh38",
    "numOfCasesInCohort": 12922,
    "occurrenceInCohort": "2 / 12922",
    "percentage": 106.02059991327963,
    "refName": "chr3",
    "referenceAllele": "C",
    "score": 2,
    "ssmId": "68e512b9-511f-58e3-8f41-137c6c0291af",
    "start": 124908159,
    "startPosition": 124908160,
    "type": "Simple Somatic Mutation",
    "uniqueId": "68e512b9-511f-58e3-8f41-137c6c0291af",
  },
  Object {
    "chromosome": "chr3",
    "consequence": Object {},
    "cosmicId": null,
    "end": 124908195,
    "endPosition": 124908195,
    "genomicDnaChange": "chr3:g.124908195C>A",
    "mutationSubtype": "Single base substitution",
    "mutationType": "Simple Somatic Mutation",
    "ncbiBuild": "GRCh38",
    "numOfCasesInCohort": 12922,
    "occurrenceInCohort": "1 / 12922",
    "percentage": 100,
    "refName": "chr3",
    "referenceAllele": "C",
    "score": 1,
    "ssmId": "0198e63f-185f-526f-96e5-f54010ffd2c0",
    "start": 124908194,
    "startPosition": 124908195,
    "type": "Simple Somatic Mutation",
    "uniqueId": "0198e63f-185f-526f-96e5-f54010ffd2c0",
  },
]
`;
